userapi/storage/postgres/localpart_external_ids_table.go

```diff
@@ -0,0 +1,97 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"github.com/element-hq/dendrite/internal/sqlutil"
+	"github.com/element-hq/dendrite/userapi/api"
+	"github.com/element-hq/dendrite/userapi/storage/tables"
+	log "github.com/sirupsen/logrus"
+)
+
+const localpartExternalIDsSchema = `
+-- Stores data about connections between accounts and third-party auth providers
+CREATE TABLE IF NOT EXISTS userapi_localpart_external_ids (
+    -- The Matrix user ID for this account
+	localpart TEXT NOT NULL,
+    -- The external ID
+	external_id TEXT NOT NULL,
+    -- Auth provider ID (see OIDCProvider.IDPID)
+	auth_provider TEXT NOT NULL,
+    -- When this connection was created, as a unix timestamp.
+    created_ts BIGINT NOT NULL,
+
+    CONSTRAINT userapi_localpart_external_ids_external_id_auth_provider_unique UNIQUE(external_id, auth_provider),
+    CONSTRAINT userapi_localpart_external_ids_localpart_external_id_auth_provider_unique UNIQUE(localpart, external_id, auth_provider)
+);
+
+-- This index allows efficient lookup of the local user by the external ID
+CREATE INDEX IF NOT EXISTS userapi_external_id_auth_provider_idx ON userapi_localpart_external_ids(external_id, auth_provider);
+`
+
+const insertUserExternalIDSQL = "" +
+	"INSERT INTO userapi_localpart_external_ids(localpart, external_id, auth_provider, created_ts) VALUES ($1, $2, $3, $4)"
+
+const selectUserExternalIDSQL = "" +
+	"SELECT localpart, created_ts FROM userapi_localpart_external_ids WHERE external_id = $1 AND auth_provider = $2"
+
+const deleteUserExternalIDSQL = "" +
+	"SELECT localpart, external_id, auth_provider, created_ts FROM userapi_localpart_external_ids WHERE external_id = $1 AND auth_provider = $2"
```

Comment:
This doesn't look like a `DELETE`. (same for SQLite)

Reply:
🤦